packages/sanity/src/core/form/types/fieldProps.ts

```diff
@@ -101,6 +107,12 @@ export type PrimitiveFieldProps = NumberFieldProps | BooleanFieldProps | StringF
 /** @beta */
 export type FieldProps =
   | ObjectFieldProps
+  | ObjectFieldProps<CrossDatasetReference>
```

Comment:
Would be nice to haved aligned `*Value` suffix on these things. So this would be `CrossDatasetReferenceValue`.